src/GraphViewer.jsx

```diff
@@ -87,663 +87,85 @@ export default class GraphViewer extends React.Component {
-                           gCobbDouglasYName={this.props.gCobbDouglasYName}
-
-                           gNName={this.props.gNName}
-                           gFunctionChoice={this.props.gFunctionChoice}
-
-                           gAreaConfiguration={this.props.gAreaConfiguration}
-                           gAreaConfigurationInitial={
-                               this.props.gAreaConfigurationInitial}
-                           gIsAreaDisplayed={this.props.gIsAreaDisplayed}
-
-                           gAreaAName={this.props.gAreaAName}
-                           gAreaBName={this.props.gAreaBName}
-                           gAreaCName={this.props.gAreaCName}
-                       />;
             rightSide = <NonLinearDemandSupplyEditor
-                            isInstructor={isInstructor}
-                            displayLabels={displayLabels}
-                            displaySliders={displaySliders}
-                            gLine1Label={this.props.gLine1Label}
-                            gLine2Label={this.props.gLine2Label}
-                            gCobbDouglasA={this.props.gCobbDouglasA}
-                            gCobbDouglasAInitial={this.props.gCobbDouglasAInitial}
-                            gCobbDouglasAName={this.props.gCobbDouglasAName}
-                            gCobbDouglasK={this.props.gCobbDouglasK}
-                            gCobbDouglasKInitial={this.props.gCobbDouglasKInitial}
-                            gCobbDouglasKName={this.props.gCobbDouglasKName}
-                            gNName={this.props.gNName}
-                            gLine1Slope={this.props.gLine1Slope}
-                            gLine1OffsetX={this.props.gLine1OffsetX}
-                            gLine1OffsetY={this.props.gLine1OffsetY}
-                            gLine2OffsetX={this.props.gLine2OffsetX}
-                            gLine2OffsetY={this.props.gLine2OffsetY}
-                            gIntersectionLabel={this.props.gIntersectionLabel}
-                            gIntersectionHorizLineLabel={this.props.gIntersectionHorizLineLabel}
-                            gIntersectionVertLineLabel={this.props.gIntersectionVertLineLabel}
-                            gFunctionChoice={this.props.gFunctionChoice}
-                            gAreaConfiguration={this.props.gAreaConfiguration}
-                            gIsAreaDisplayed={this.props.gIsAreaDisplayed}
-                            gAreaAName={this.props.gAreaAName}
-                            gAreaBName={this.props.gAreaBName}
-                            gAreaCName={this.props.gAreaCName}
                             showAUC={this.props.gType === 10}
-                            updateGraph={this.updateGraph}
+                            {...commonViewerProps}
+                            {...this.props}
                         />;
         } else if (this.props.gType === 3 || this.props.gType === 12) {
-            leftSide = <JXGBoard
```

Comment:
I'm getting a JS error when testing these changes with this graph type, "Input-Output Illustrations". I can create the graph via GraphPicker/Editor, but then when I save and view it there it can't render:
```
viewer.js:461 Uncaught TypeError: Cannot read properties of null (reading 'create')
    at NonLinearDemandSupplyGraph.make (viewer.js:461:500)
    at mkNonLinearDemandSupply (viewer.js:462:205)
    at JXGBoard.renderJXBoard (viewer.js:723:87)
    at JXGBoard.componentDidUpdate (viewer.js:725:3672)
    at commitLayoutEffectOnFiber (viewer.js:60838:30)
    at commitLayoutMountEffects_complete (viewer.js:62193:9)
    at commitLayoutEffects_begin (viewer.js:62179:7)
    at commitLayoutEffects (viewer.js:62117:3)
    at commitRootImpl (viewer.js:64328:5)
    at commitRoot (viewer.js:64187:5)
```

I'm not sure exactly where the error is caused yet, as these are some pretty extensive changes.

Reply:
I'm afraid I can't replicate the error. Were there any more details?

Reply:
I'll take a look and see if I can find what's going on here.

Reply:
The call stack points to `this.board` being null in `NonLinearDemandSupplyGraph.make()` in Graph.js. The refactor here removes the custom JXGBoard defined in this if clause and instead uses a common JXGBoard defined on line 100, which should be fine. There must be something in the changes here preventing the graph builder from hooking up to the jsxgraph board, because this error isn't occurring on the master branch. I'll dig deeper here.

Reply:
My current theory on this is that your cleanup code actually uncovered a pre-existing race condition bug in JXGBoard - which is a great thing! I think we are making the code work better in addition to being clearer.

It looks like the code that creates JXGBoard.board2 for this graph type is being called in didComponentMount, when the gType is still null, so this second board for this special graph type is never created here. I will refactor things to fix this.

Reply:
I've added a new commit to this branch which fixes the error I was seeing - see my commit notes for details.